torch_xla/distributed/fsdp/xla_fully_sharded_data_parallel.py

```diff
@@ -1471,48 +1478,58 @@ def get_original_names_and_sharded_parameters(self):
 
     return orig_named_parameters
 
-  def get_shard_metadata(self):
+  def load_state_dict(self, state_dict, strict=True):
+    if "shard_metadata" in state_dict:
+      state_dict.pop("shard_metadata")
+    super().load_state_dict(state_dict, strict=strict)
+
+  @staticmethod
+  def get_shard_metadata(m, destination, prefix, local_metadata):
     """
     Get the shard metadata to consolidate the sharded model checkpoints.
-    The output from this method should be saved in a checkpoint file and
-    used in ``consolidate_sharded_model_checkpoints``.
+    The output from this method will be used in ``consolidate_sharded_model_checkpoints``.
     """
     shard_info = {}
     flatten_info = {}
     buffer_info = {}
-    for module_name, m in self.named_modules():  # includes self
```

Comment:
I wonder would the `lm_head.lm_head` example in this PR be fixed if we change this line to 
```
for module_name, m in self.named_modules(remove_duplicate=False): # includes self
```